numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -383,6 +383,36 @@ npy_PyFile_CloseFile(PyObject *file)
 }
 
 
+/* This is a copy of _PyErr_ChainExceptions
+ */
+static NPY_INLINE void
+npy_PyErr_ChainExceptions(PyObject *exc, PyObject *val, PyObject *tb)
+{
+    if (exc == NULL)
+        return;
+
+    if (PyErr_Occurred()) {
+        /* only py3 supports this anyway */
+        #ifdef NPY_PY3K
+            PyObject *exc2, *val2, *tb2;
```

Comment:
This implementation has been taken from the cpython source: https://github.com/python/cpython/blob/50b48572d9a90c5bb36e2bef6179548ea927a35a/Python/errors.c#L389-L414